pkg/services/ssosettings/strategies/oauth_strategy.go

```diff
@@ -51,13 +51,20 @@ func (s *OAuthStrategy) loadAllSettings() {
 	allProviders := append(ssosettings.AllOAuthProviders, social.GrafanaNetProviderName)
 	for _, provider := range allProviders {
 		settings := s.loadSettingsForProvider(provider)
-		if provider == social.GrafanaNetProviderName {
+		// This is required to support the legacy settings for the provider (auth.grafananet section)
+		// It will use the settings (and overwrite the current grafana_com settings) from auth.grafananet if
+		// the auth.grafananet section is enabled and the auth.grafana_com section is disabled.
+		if provider == social.GrafanaNetProviderName && s.shouldUseGrafanaNetSettings(settings) {
 			provider = social.GrafanaComProviderName
 		}
 		s.settingsByProvider[provider] = settings
 	}
 }
 
+func (s *OAuthStrategy) shouldUseGrafanaNetSettings(settings map[string]any) bool {
```

Comment:
This func is a bit confusing because the settings param are not necessarily referring to grafanaNet as the func name suggests, the settings can be any settings from any provider.

I suggest you rename this func to something like `isGrafanaComDisabled()` with the following body `return s.settingsByProvider[social.GrafanaComProviderName]["enabled"] == false` and then move the condition `settings["enabled"] == true` to the line 57.

Reply:
Thanks, done!